pipeline-runner/R/gem2s-4-score_doublets.R

```diff
@@ -1,68 +1,65 @@
-#  - Compute doublet scores per sample
-#  - Save an txt file with doublet scores per sample [doublet-scores-sampleName]
-
-score_doublets <- function(input, pipeline_config) {
-  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
-  message("reloading old matrices...")
-
-  scdata_list <- readRDS("/output/pre-doublet-scdata_list.rds")
-
-  message("Loading configuration...")
-  config <- RJSONIO::fromJSON("/input/meta.json")
-
-  print_config(4, "Score Doublets", input, pipeline_config, config)
-
-  # Check which samples have been selected. Otherwiser we are going to use all of them.
-  if (length(config$samples) > 0) {
-    samples <- config$samples
-  } else {
-    samples <- names(scdata_list)
-  }
-
-  scdata_list <- scdata_list[samples]
-
-  message("calculating probability of barcodes being doublets...")
-  for (sample_name in names(scdata_list)) {
-    compute_doublet_scores(scdata_list[[sample_name]], sample_name)
+#' Compute doublet score for count matrices
+#'
+#' @inheritParams download_cellranger
+#' @param prev_out 'output' slot from call to \code{run_emptydrops}
+#'
+#' @return \code{prev_out} with added slot 'doublet_scores' containing result of call to
+#'   \code{\link{compute_doublet_scores}} for each sample.
+#'
+#' @export
+#'
+score_doublets <- function(input, pipeline_config, prev_out) {
```

Comment:
again small recommendation: make the title of this function consistent with what you have for empty drops: call it `run_doublet_scores`

Reply:
The GEM2S function names don't follow a convention that they begin with `run_`. For example we have `load_cellranger_files`, `create_seurat`, `prepare_experiment`, `upload_to_aws`.

Reply:
I would consider introducing a little convention, because it can be confusing which function is being the one called and which is the one internal to the file. 
I would also recommend defining the ones that are only internally used as private, but I don't think you can have private methods in R without having classes. Or can you?

Reply:
I thought it would be a pain, this is why I didn't really suggest it. But ideally when you have a couple of methods, some of them are only intended for internal use, it is good to restrict them so that in the future they don't get misused and also so that the code is more readable. It is not a huge deal though

Reply:
For GEM2S all the main functions that are also endpoints are the only functions that are exported. All other GEM2S utilities/helpers don't have the `@export` tag so if the R code is installed as a package they are internal. 

As Oliver mentioned though we do not install the package when we run it in a docker container but instead source all the R code, making everything available. The upside of doing it this way is that the watcher works (the pipeline hot reloads).

Reply:
I see, sounds reasonable